src/lazy/expanded/template.rs

```diff
@@ -899,46 +892,34 @@ impl TemplateBodyExprKind {
 }
 
 /// A macro invocation found in the body of a template.
-///
-/// Because the template definition lives in the macro table (which may need to grow in the process
-/// of evaluating a stream), this type holds the address of the invoked macro rather than a
-/// reference to it.
```

Comment:
🗺️ Now that we're using `Rc`, the template definition's dependent data will live as long as necessary. And because macros cannot have cyclical dependencies, the `Rc`s cannot accidentally form cycles either.

Reply:
Also, if the macro `struct` is immutable, that's another guarantee that there will be no `Rc` cycles because you can't create something with a reference to something else that doesn't exist yet.